#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_amd64-disable-platform-reset.dpatch by Robert Millan <rmh@aybabtu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: See these mailing lists post for a description of this patch:
## DP: http://lists.alioth.debian.org/pipermail/pkg-boinc-devel/2006-August/000352.html
## DP: http://lists.alioth.debian.org/pipermail/pkg-boinc-devel/2006-August/000350.html

@DPATCH@
diff -urNad boinc-5.4.11~/client/client_state.C boinc-5.4.11/client/client_state.C
--- boinc-5.4.11~/client/client_state.C	2006-08-03 10:31:03.000000000 +0200
+++ boinc-5.4.11/client/client_state.C	2006-08-26 14:01:36.000000000 +0200
@@ -331,6 +331,9 @@
     // This could happen e.g. if user copies files from PPC to Intel Mac
     //
     if (strcmp(platform_name, statefile_platform_name.c_str())) {
+#ifdef __x86_64__
+        platform_name = strdup(statefile_platform_name.c_str());
+#else
         msg_printf(NULL, MSG_INFO,
             "Platform changed from %s to %s - resetting projects",
             statefile_platform_name.c_str(), platform_name
@@ -339,6 +342,7 @@
             p = projects[i];
             reset_project(p);
         }
+#endif
     }
 
     return 0;
